java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -84,4 +90,107 @@ public boolean isSubtype(JavaType t, JavaType s) {
     return result;
   }
 
+  /**
+   * Compute the "Least Upper Bound" ("lub", jls8 §4.10.4) of a list of type. The "lub" is a shared supertype that is more specific than any
+   * other shared supertype (that is, no other shared supertype is a subtype of the least upper bound)
+   *
+   * FIXME Parameterized types are currently ignored, as the method is used only to handle Union Types Trees, themselves being used only
+   * in catch trees. Note that Exceptions (any subclass of Throwable) cannot be generic (jls8 §8.1.2, §11.1.1: "compile-time error if a generic
+   * class is a direct or indirect subclass of Throwable")
+   *
+   * @param types
+   * @return the least upper bound of the types
+   */
+  public Type leastUpperBound(Type... types) {
+    Preconditions.checkArgument(types.length > 1);
+
+    List<List<Type>> supertypes = supertypes(types);
+    if (containsParameterizedTypes(supertypes)) {
+      // Parameterized types are ignored, erased supertypes should be calculated
+      return Symbols.unknownType;
+    }
+
+    List<Type> candidates = intersection(supertypes);
+    List<Type> minimalCandidates = minimalCandidates(candidates);
+    if (minimalCandidates.isEmpty()) {
+      return Symbols.unknownType;
+    }
+
+    // FIXME huge approximation: should be the bound of all the minimalCandidates, not only the first type
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)